src/language/concurrency.md

```diff
@@ -235,305 +268,268 @@ After handling the events, the isolate exits.
+performance optimizations such as sharing code; a new isolate immediately runs
+the code owned by the isolate group. Also, `Isolate.exit()` works only when the
+isolates are in the same isolate group.
 
-### Sending multiple messages between isolates
+In some special cases, you might need to use [`Isolate.spawnUri()`][], which
+sets up the new isolate with a copy of the code that's at the specified URI.
+However, `spawnUri()` is much slower than `spawn()`, and the new isolate isn't
+in its spawner's isolate group. Another performance consequence is that message
+passing is slower when isolates are in different groups.
 
-`Isolate.run()` abstracts a handful of lower-level, 
-isolate-related API to simplify isolate management:
+[`Isolate.spawnUri()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawnUri.html
 
-* [`Isolate.spawn()`][] and [`Isolate.exit()`][]
-* [`ReceivePort`][] and [`SendPort`][]
+## Limitations of isolates
+
+Isolates are most commonly used in Flutter applications, when you need to do
+large computations that would cause the UI to become unresponsive, but they’re
+also useful in many server-side applications. In general, isolates are useful
+whenever your application is handling computations that are large enough to
+temporarily block other computations.
+
+While there isn’t one hard-and-fast rule about when to use isolates, there are
+common situations in which you can consider using isolates. The most common uses
+for isolates are:
+
+- Parsing and decoding exceptionally large JSON blobs
+- Processing and compressing photos, audio and video
+- Converting audio and video files
+- Performing complex searching and filtering on large lists or within
+  filesystems
+- Performing I/O, such as communicating with a database
+- Handling a large volume of network requests
+
+### Isolates aren't threads
+
+If you’re coming to Dart from a language with multithreading, it’d be reasonable
+to expect isolates to behave like threads, but that isn’t the case. Isolates
+have their own memory heaps, ensuring that none of the state in an isolate is
+accessible from any other isolate. Therefore, isolates are limited by their
+access to their own memory. For example, if you have an application with a
+global mutable variable called `configuration`, that variable will be a separate
+variable in your spawned isolate. If you mutate that variable in the spawned
+isolate, it will remain untouched in the main isolate. This is how isolates are
+meant to function, and it's important to keep in mind when you’re considering
+using isolates.
+
+### Message types
+
+`message`s sent via SendPort can be almost any type of Dart object, but there
+are a few exceptions:
+
+- Objects with native resources (subclasses of e.g. NativeFieldWrapperClass1). A
+  Socket object for example refers internally to objects that have native
+  resources attached and can therefore not be sent.
```

Comment:
```suggestion
- Objects with native resources
  - For example, subclasses of [`NativeFieldWrapperClass1`](), or a
    [`Socket`]() object that refers internally to objects that have native
    resources attached and can therefore not be sent.
```

I don't understand the given example(s ?), but basically just keep any elaboration in a sub-bullet and code font + link on any API mentions